### PR TITLE
Do not require model class in paginated relations

### LINKED ISSUE #1001: Possible BC 4.4: Failed model detection
I am not 100% sure what is exactly causing the issue but on 4.3 we have the following schema:

```graphql
type SchoolGroup {
    children: [SchoolGroup!] @hasMany(type: "paginator")
}
```

However when we just tried updating to 4.4 the following error shows up:

```
No class 'SchoolGroup' was found for directive 'hasMany'
```

(which is true, `SchoolGroup` is not an actual class/model, it's namespaced in `...\Models\School\Group` in our app)

This can be fixed with adding the `@modelClass` directive it looks like, but this was never a problem before so I'm puzzled what happened and what the exact way to fix this is, it always just worked :tm: :slightly_smiling_face:

As far as I could tell a `4.3`->`4.4` is not supposed to break existing schema's so this might be a bug. I have tried reading through the PR's that seem to be causing them but they're massive, so I'm not able to determine the root cause exactly.

### PR DESCRIPTION
**Pull request by spawnia**
- [x] Added or updated tests
- [ ] Added Docs for all relevant versions
- [x] Updated the changelog

Fixes #1001 

**Changes**

Fix regression in 4.4.0 that required matching the type returned from paginated relationship fields with the class name of the model

**Breaking changes**

Actually unbreaking a regression

### CODE REVIEW

File: src/Pagination/PaginationManipulator.php [hunk 1]
```diff
@@ -137,7 +153,7 @@ protected function registerPaginator(
         $paginatorFieldClassName = addslashes(PaginatorField::class);
 
         $paginatorType = PartialParser::objectTypeDefinition("
-            type $paginatorTypeName $modelClassDirective {
+            type $paginatorTypeName {$this->modelClassDirective()} {
```
Comment (joskfg) [inline src/Pagination/PaginationManipulator.php]: Just a tip:

If you use HEREDOC, you won't need to escape all that quotes.

File: tests/Integration/Schema/Directives/HasManyDirectiveTest.php [hunk 2]
```diff
@@ -158,6 +158,52 @@ public function testCanQueryHasManyPaginator(): void
         ])->assertJsonCount(2, 'data.user.tasks.data');
     }
 
+    public function testDoesNotRequireModelClassForPaginatedHasMany(): void
```
Comment (joskfg) [inline tests/Integration/Schema/Directives/HasManyDirectiveTest.php]: I don't see the difference between this test and the previous one. Both doesn't have the modelClass set, just the type.

Comment (spawnia) [reply tests/Integration/Schema/Directives/HasManyDirectiveTest.php]: The type name does not match the model name. It is not needed, the BC was that 4.4 forced them to line up.

Renamed to `type NotTheModelNameTask` for clarification.

File: src/Pagination/PaginationManipulator.php (hunk 1 above)
Comment (spawnia) [reply src/Pagination/PaginationManipulator.php]: Let's discuss this in a seperate PR, that could be useful in a lot of places!

**Pull request merged**